Build docs PR preview on RTD

stevepiercy opened the pull request:
Closes #106.
Closes #110.
Closes #117.


<!-- readthedocs-preview sphinxcontrib-httpexample start -->
----
📚 Documentation preview 📚: https://sphinxcontrib-httpexample--115.org.readthedocs.build/

<!-- readthedocs-preview sphinxcontrib-httpexample end -->

datakurre commented:
Yes. Thanks. It’s good for me learn these gaps. In theory, the setup should use uv.lock transpatently to support both worlds, but there are always devils in the details.

stevepiercy commented on Makefile:
```diff
@@ -7,12 +7,17 @@ SPHINXOPTS      ?=
+ifdef IN_NIX_SHELL
+BIN_FOLDER      = $(shell dirname "$$(command -v python)")
+VENV_FOLDER     = $(shell dirname "$(BIN_FOLDER)")
+else
```
TIL! Nifty!

stevepiercy reviewed (commented):
Everything LGTM. I checked it out locally, and was able to run the make commands.

stevepiercy commented:
@datakurre I'm still not absolutely clear what to do, but I made a few changes. The example is for `@plone/client`, so I refactored it accordingly, and added some docs.

Then I looked at the relevant issues and PRs, but I'm not sure what you have in mind.

- https://github.com/collective/sphinxcontrib-httpexample/pull/92
- https://github.com/plone/volto/pull/5576

@sneridagh @robgietema can you help fill in details of what you'd like to see?

Please let me know. Thank you!

stevepiercy commented:
@datakurre thanks. That's not too bad. I still want to drop Python 3.9 from support, so I'll update https://github.com/collective/sphinxcontrib-httpexample/pull/115/changes/4fd0867e1f9a72dcf1e2ec20ba49df3e6c24d1d2#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R74.

stevepiercy commented:
@Hrittik20 sounds promising. Please give it a try.

I tried several times to run a debugging session with break points to see what was actually getting passed into the classes on instantiation. I couldn't make sense of it. Maybe I was tired or looking at the wrong thing. If you or @datakurre have an idea of how to do so, I'd like to hear suggestions.

stevepiercy commented:
@datakurre example at https://sphinxcontrib-httpexample--115.org.readthedocs.build/en/115/usage.html#result.

Toggle the `http` and `response` tabs.

<img width="723" height="201" alt="Screenshot 2026-02-18 at 2 11 07 AM" src="https://github.com/user-attachments/assets/2d525469-e2a3-4cc3-a597-b38ff6950911" />

<img width="723" height="201" alt="Screenshot 2026-02-18 at 2 10 35 AM" src="https://github.com/user-attachments/assets/85a38ac7-2bc8-434b-a596-3722bdaa494b" />

The tab library examples don't have this issue.

stevepiercy commented:
@datakurre thanks for the workaround and troubleshooting.

The lexer error you mention in Sphinx is disappointing. In that case, should I remove the claim of support for `HTTP ` (trailing space without version) from both the documentation and the code?

https://sphinxcontrib-httpexample--115.org.readthedocs.build/en/115/usage.html#http-with-space-no-version

https://github.com/collective/sphinxcontrib-httpexample/blob/b0082de6ce9db4e9402b24961757a00d79cb6c5b/src/sphinxcontrib/httpexample/directives.py#L58-L78

FWIW, the Pygments lexer for `http` works with `HTTP OK 200` in [this demo](https://pygments.org/demo/?lexer=http&style=staroffice&formatter=html&code=GET+%2FPlone%2Ffront-page+HTTP%2F1.1%0AHost%3A+localhost%3A8080%0AAccept%3A+application%2Fjson%0AAuthorization%3A+Basic+YWRtaW46YWRtaW4%3D%0A%0A%0AHTTP+200+OK%0AContent-Type%3A+application%2Fjson%0A%0A%7B%0A++%22%40id%22%3A+%22http%3A%2F%2Flocalhost%3A8080%2FPlone%2Ffront-page%22%2C%0A++%22%40type%22%3A+%22Document%22%2C%0A++%22UID%22%3A+%221f699ffa110e45afb1ba502f75f7ec33%22%2C%0A++%22allow_discussion%22%3A+null%2C%0A++%22changeNote%22%3A+%22%22%2C%0A++%22contributors%22%3A+%5B%5D%2C%0A++%22created%22%3A+%222016-01-21T01%3A14%3A48%2B00%3A00%22%2C%0A++%22creators%22%3A+%5B%0A++++%22test_user_1_%22%2C%0A++++%22admin%22%0A++%5D%2C%0A++%22description%22%3A+%22Congratulations%21+You+have+successfully+installed+Plone.%22%2C%0A++%22effective%22%3A+null%2C%0A++%22exclude_from_nav%22%3A+false%2C%0A++%22expires%22%3A+null%2C%0A++%22id%22%3A+%22front-page%22%2C%0A++%22language%22%3A+%22%22%2C%0A++%22modified%22%3A+%222016-01-21T01%3A24%3A11%2B00%3A00%22%2C%0A++%22parent%22%3A+%7B%0A++++%22%40id%22%3A+%22http%3A%2F%2Flocalhost%3A8080%2FPlone%22%2C%0A++++%22%40type%22%3A+%22Plone+Site%22%2C%0A++++%22description%22%3A+%22%22%2C%0A++++%22title%22%3A+%22Plone+site%22%0A++%7D%2C%0A++%22relatedItems%22%3A+%5B%5D%2C%0A++%22review_state%22%3A+%22private%22%2C%0A++%22rights%22%3A+%22%22%2C%0A++%22subjects%22%3A+%5B%5D%2C%0A++%22table_of_contents%22%3A+null%2C%0A++%22text%22%3A+%7B%0A++++%22content-type%22%3A+%22text%2Fplain%22%2C%0A++++%22data%22%3A+%22If+you%27re+seeing+this+instead+of+the+web+site+you+were+expecting%2C+the+owner+of+this+web+site+has+just+installed+Plone.+Do+not+contact+the+Plone+Team+or+the+Plone+mailing+lists+about+this.%22%2C%0A++++%22encoding%22%3A+%22utf-8%22%0A++%7D%2C%0A++%22title%22%3A+%22Welcome+to+Plone%22%0A%7D).

There's only two examples in `plone.restapi` that use `HTTP ` and the rest use `HTTP/`, but these files appear to be orphans. They're not linked in the docs, and it appears they're not even used anymore.

- https://github.com/plone/plone.restapi/blob/bc7fc79d447d625fe90696285b6b425a83a760be/docs/source/endpoints/copymove.md
- https://github.com/plone/plone.restapi/blob/bc7fc79d447d625fe90696285b6b425a83a760be/src/plone/restapi/tests/http-examples/move.json
- https://github.com/plone/plone.restapi/blob/bc7fc79d447d625fe90696285b6b425a83a760be/src/plone/restapi/tests/http-examples/copy.json

Please let me know. Thank you!

stevepiercy commented on tests/test_lexers.py:
```diff
@@ -0,0 +1,130 @@
+
+
+def test_minimal_response(lexer):
+    """Test that a minimal response is lexed correctly."""
```
It would be nice to parametrize this test, and possibly others, for all three formats of the response header mentioned in https://github.com/collective/sphinxcontrib-httpexample/pull/115/changes/1464691ff4ac0a324e37453d89b276d47ae040c9..8134d5851798ebc78da24267aa0f093cb6657247#diff-963b5a08cd1975063a52723b1cd43303e4c663b7259b8920a41fe4f92d1df7f2R46.

Hrittik20 commented:
@stevepiercy LGTM from my side. Glad the response-tab bug got fixed!
